src/dependency-version-checker.js

```diff
@@ -55,11 +58,28 @@ class DependencyVersionChecker {
     }
 
     if (!this.isAbove(compareVersion)) {
-      let error = new Error(message);
+      throw new SilentError(message);
+    }
+  }
 
-      error.suppressStacktrace = true;
+  isSingleton() {
+    if (this._isSingleton) {
+      return this._isSingleton;
+    }
+    let allInstances = discoverAddons(
```

Comment:
rather then finding all instances, should we adjust this algo bail early when more than once is found? For small projects this shouldn't matter, but voyager-web size projects it could.